source/agora/node/Config.d

```diff
@@ -170,15 +175,74 @@ public struct InterfaceConfig
      /// Bind address
     public string address;
 
-    /// Bind port
-    public ushort port;
+    /***************************************************************************
+
+        Bind port, default to the scheme's default
+
+        Default values are 80 for `http`, 443 for `https`, and 2826 for `tcp`.
+
+    ***************************************************************************/
+
+    public @Optional ushort port;
+
+    /***************************************************************************
+
+        The public address of this interface
+
+        If this binds to an interface that is not publicly available,
+        this value needs to be set for the node to correctly register itself
+        on the registry.
+
+    ***************************************************************************/
+
+    public string hostname;
```

Comment:
A little bit confusing name since it can also include path for `http` schema

Reply:
Suggestions ? `public_address` ? Verbose but correct.

Reply:
I'm thinking why we need this. I thought a node should be accessible (through its interface) from all addresses that it registers. Why binding with an address at interface config?

Reply:
`hostname` should really just be an override. Ideally, what would happen is:
- User doesn't specify anything and get the default interface(s);
- If user specify something, it's what interface to bind to / what port;
- The node detects this and uses the IP address and configured port to register;

However, there are some cases where it doesn't work:
- Node is behind some sort of proxy: Nginx, a docker container, etc... And it cannot know its final IP / port (our case);
- The user wants to register a domain, not just the IP address;

We could potentially detect the domain via the reverse lookup, but there could be multiple domains pointing to the same host (like us) and virtual host matching.

That's why the override is in place, so the user can have the final say if their config doesn't match the default.